nntrainer/layers/lstmcell.cpp

```diff
@@ -23,365 +25,315 @@ namespace nntrainer {
 
 static constexpr size_t SINGLE_INOUT_IDX = 0;
 
-enum LSTMParams {
-  weight_xh,
+enum LSTMCellParams {
+  weight_ih,
   weight_hh,
-  bias_h,
+  bias_ih,
   hidden_state,
-  mem_cell,
-  fgio,
+  cell_state,
+  ifgo,
   dropout_mask
 };
 
+const std::vector<unsigned int>
+getInOutIdx(std::array<unsigned int, 7> &wt_idx) {
+  std::vector<unsigned int> ret(3);
+  ret[0] = SINGLE_INOUT_IDX;
+  ret[1] = wt_idx[LSTMCellParams::hidden_state];
+  ret[2] = wt_idx[LSTMCellParams::cell_state];
+  return ret;
+}
+
+const std::vector<unsigned int>
+getTensorIdx(std::array<unsigned int, 7> &wt_idx) {
+  std::vector<unsigned int> ret(1);
+  ret[0] = wt_idx[LSTMCellParams::ifgo];
+  return ret;
+}
+
 LSTMCellLayer::LSTMCellLayer() :
   LayerImpl(),
-  lstm_props(props::Unit(), props::HiddenStateActivation(),
-             props::RecurrentActivation(), props::DropOutRate(),
-             props::MaxTimestep(), props::Timestep()),
+  lstmcell_props(props::Unit(), props::DropOutRate(), props::MaxTimestep(),
+                 props::Timestep()),
   wt_idx({0}),
-  acti_func(ActivationType::ACT_NONE, true),
-  recurrent_acti_func(ActivationType::ACT_NONE, true),
   epsilon(1e-3) {}
 
-// - weight_xh ( input to hidden )
-//  : [1, 1, input_size, unit (hidden_size) x NUM_GATE] -> f, g, i, o
-// - weight_hh ( hidden to hidden )
-//  : [1, 1, unit (hidden_size) , unit (hidden_size) x NUM_GATE] -> f, g, i, o
-// - bias_h ( hidden bias )
-//  : [1, 1, 1, unit (hidden_size) x NUM_GATE] -> f, g, i, o
 void LSTMCellLayer::finalize(InitLayerContext &context) {
-  auto &weight_regularizer =
+  NNTR_THROW_IF(std::get<props::Unit>(lstmcell_props).empty(),
+                std::invalid_argument)
+    << "unit property missing for lstmcell layer";
+  const unsigned int unit = std::get<props::Unit>(lstmcell_props).get();
+  const float dropout_rate = std::get<props::DropOutRate>(lstmcell_props);
+  const unsigned int max_timestep =
+    std::get<props::MaxTimestep>(lstmcell_props);
+
+#if !(ENABLE_SHARING_WT_IDX)
```

Comment:
later, please add a comment what this signifies.

Reply:
Added